dali/python/backend_impl.cc

```diff
@@ -222,6 +222,34 @@ void ExposeTensor(py::module &m) {
       String representing NumPy type of the Tensor.
       )code");
 }
+
+template <typename Backend>
+std::unique_ptr<Tensor<Backend> > TensorListGetItemImpl(TensorList<Backend> &t, Index id) {
+  if (id < 0) {
+    int num_tensors = static_cast<int>(t.ntensor());
+    if (id < -num_tensors)
+      throw py::index_error("TensorListCPU index out of range");
+    id = num_tensors + id;
+  }
+  std::unique_ptr<Tensor<Backend>> ptr(new Tensor<Backend>());
+  ptr->ShareData(&t, id);
+  return ptr;
+}
+
+template <typename Backend>
+py::list TensorListGetItemSliceImpl(TensorList<Backend> &t, py::slice slice) {
+  size_t start, stop, step, slicelength;
+  if (!slice.compute(t.ntensor(), &start, &stop, &step, &slicelength))
+      throw py::error_already_set();
+  py::list list;
```

Comment:
How does it work with advanced indexing, like `list[:]`, `list[-1]`, `list[1:]`, etc?